python/freetensor/core/transformer.py

```diff
@@ -301,6 +302,22 @@ def assert_stmt(test):
         assert test
 
 
+def load_attr(obj, attr: str):
+    '''Load attribute staging tool. Allows customization of reading attributes.'''
+    try:
+        return getattr(obj, attr)
+    except:
```

Comment:
Maybe a specific exception type? Generic exception is hell for debugging

Reply:
`load_attr` not only covers special attributes but also other user-defined attributes. If `a.b` is written but `a` has no attribute `b`, it should raise an `AttributeError` as is expected in normal Python programming, instead of some FreeTensor -defined error I think.